### ISSUE TITLE
bug (?): xh removes the content-encoding header

### ISSUE THREAD

**Issue by jbr** [ducaale/xh#210]
Hi, and thanks for xh! I've been using it to develop a server framework, and noticed that although xh does a great job of automatically processing gzip and brotli encoded response bodies, it seems to _remove_ the Content-Encoding header from verbose display. If this is in fact a bug, would you like a patch PR?

**Comment by blyxxyz** [comment]
It's not intentional on our part, and HTTPie doesn't do it. So I'd say it's a bug. PR welcome!

It seems to happen here, in reqwest: https://github.com/seanmonstar/reqwest/blob/baffb9c004fbd4fb235046683edb3f86ab74596b/src/async_impl/decoder.rs#L172

Which is called unconditionally from here: https://github.com/seanmonstar/reqwest/blob/baffb9c004fbd4fb235046683edb3f86ab74596b/src/async_impl/response.rs#L50

So I'm not sure how to go about fixing it.

**Comment by ducaale** [comment]
For reference, https://github.com/seanmonstar/reqwest/pull/68 is the PR that introduced the removal of `Content-Encoding`, `Content-Length`, and `Accept-Encoding` whenever the response body is compressed.

**Comment by ducaale** [comment]
I've opened https://github.com/seanmonstar/reqwest/pull/1413 for opting out from the behavior introduced by https://github.com/seanmonstar/reqwest/pull/68

**Comment by ducaale** [comment]
For now, I think the best solution would be to replace reqwest's auto-decompression with our own code in a similar fashion to how we [decode responses](https://github.com/ducaale/xh/blob/develop/src/printer.rs#L561) now.

```rs
use flate2::read::GzDecoder;

fn decompress_blob<'a>(raw: &'a [u8], content_encoding: &str) -> anyhow::Result<Cow<'a, [u8]>> {
    match content_encoding {
        "gzip" => {
            let mut decoder = GzDecoder::new(raw);
            let mut buf: Vec<u8> = Vec::new();
            decoder.read_to_end(&mut buf)?;
            Ok(Cow::from(buf))
        }
        "deflate" => todo!(),
        "brotli" => todo!(),
        _ => Ok(Cow::from(raw)),
    }
}

fn decompress_stream<'a>(response: &'a mut impl Read, content_encoding: &str) -> io::Result<impl Read + 'a> {
    todo!()
}
```

**Comment by ducaale** [comment]
This is now fixed in xh v0.16.0
